contract-tests/test_api.py

```diff
@@ -121,17 +115,11 @@ def test_action_signatures(conf, requests_session):
     if len(data) == 0:
         pytest.skip("No signed actions")
 
-    cert_urls = set()
-
     for item in data:
         canonical_action = canonical_json(item["action"])
         signature = item["signature"]["signature"]
-        pubkey = item["signature"]["public_key"]
-        cert_urls.add(item["signature"]["x5u"])
-        assert signing.verify_signature(canonical_action, signature, pubkey)
-
-    for url in cert_urls:
-        signing.verify_x5u(url)
+        x5u = item["signature"]["x5u"]
+        assert signing.verify_signature_x5u(canonical_action, signature, x5u)
```

Comment:
This is going to cause a lot more outbound network traffic. Of the dozens of signed objects we have, there are only a few unique `x5u` urls. I wonder if it is possible for `verify_signature_x5u` to cache the response from the server, or to use the etag information to do a "if not modified" request?

```
$ http GET 'https://normandy.cdn.mozilla.net/api/v1/recipe/signed/?enabled=1' | jq '.[].recipe.id' | sort | uniq | wc -l
40

$ http GET 'https://normandy.cdn.mozilla.net/api/v1/recipe/signed/?enabled=1' | jq '.[].signature.x5u' | sort | uniq | wc -l
1
```

Reply:
While this is true that this is going to produce more requests and load, it seems like a relatively low priority for a set of contract tests. These aren't on the hot path..

Reply:
Apologies, I thought that these were the heartbeat tests. This is fine.